src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3327,13 +3318,14 @@ protected override bool HasChildItems(string path)
             path = NormalizePath(path);
 
             // First check to see if it is a directory
-
             try
             {
                 DirectoryInfo directory = new DirectoryInfo(path);
 
                 // If the above didn't throw an exception, check to
-                // see if it contains any directories
+                // see if we should proceed and if it contains any children
```

Comment:
The `DirectoryInfo` constructor will succeed if the given path does not exist or if it actually refers to a file. In the former case the `Exists` property will be false, in the latter the `Attributes` property will lack the `Directory` attribute.

The myriad other ways the constructor will fail are explicitly called out in the various `catch` blocks.